src/query/functions/binary/common.go

```diff
@@ -80,10 +80,23 @@ var (
 	errNoMatching              = errors.New("vector matching parameters must be provided for binary operations between series")
 )
 
-func tagMap(t models.Tags) map[string]models.Tag {
-	m := make(map[string]models.Tag, t.Len())
+type tags models.Tag
+
+func toTagMap(t models.Tags) *tagMap {
```

Comment:
does this show up on the flamegraph ? This is done only once per series I assume so I'm not sure if we should be optimizing at the cost of simplicity.